src/Authentication.Abstractions/AzureSession.cs

```diff
@@ -234,6 +237,7 @@ public void RegisterComponent<T>(string componentName, Func<T> componentInitiali
                 () =>
                 {
                     var key = new ComponentKey(componentName, typeof(T));
+                    // todo: double check if dictionary operations are thread safe
```

Comment:
What does this comment here mean? Anything left undone?

Reply:
might need to replace `containskey` and assign by index with atomic methods

Reply:
Updated the implementation